src/mainboard/protectli/vault_adl_n/variants/vp2430/gpio.c

```diff
@@ -0,0 +1,1415 @@
+
+	/* GPP_B12 - SLP_S0# */
+	/* DW0: 0x44000700, DW1: 0x0003c000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	/* PAD_CFG_NF(GPP_B12, NONE, DEEP, NF1), */
+	_PAD_CFG_STRUCT(GPP_B12, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE), PAD_IOSSTATE(IGNORE)),
+
+	/* GPP_B13 - PLTRST# */
+	/* DW0: 0x44000700, DW1: 0x0003c000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	/* PAD_CFG_NF(GPP_B13, NONE, DEEP, NF1), */
+	_PAD_CFG_STRUCT(GPP_B13, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE), PAD_IOSSTATE(IGNORE)),
+
+	/* GPP_B14 - SPKR */
+	/* DW0: 0x84000700, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	/* PAD_CFG_NF(GPP_B14, NONE, PLTRST, NF1), */
+	_PAD_CFG_STRUCT(GPP_B14, PAD_FUNC(NF1) | PAD_RESET(PLTRST) | PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE), 0),
+
+	/* GPP_B15 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	PAD_NC(GPP_B15, NONE),
+	PAD_NC(GPP_B16, NONE),
+	PAD_NC(GPP_B17, NONE),
+	/* GPP_B18 - GPIO */
+	/* DW0: 0x44000200, DW1: 0x00000000 */
+	PAD_CFG_GPO(GPP_B18, 0, DEEP),
+
+	/* GPP_B19 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	PAD_NC(GPP_B19, NONE),
+
+	/* GPP_B20 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	PAD_NC(GPP_B20, NONE),
+
+	/* GPP_B21 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	PAD_NC(GPP_B21, NONE),
+
+	/* GPP_B22 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	PAD_NC(GPP_B22, NONE),
+
+	/* GPP_B23 - GPIO */
+	/* DW0: 0x44000200, DW1: 0x00000000 */
+	PAD_CFG_GPO(GPP_B23, 0, DEEP),
+
+	/* GPP_B24 - GSPI0_CLK_LOOPBK */
+	/* DW0: 0x40000700, DW1: 0x00000000 */
+	/* DW0: PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	/* PAD_CFG_NF(GPP_B24, NONE, DEEP, NF1), */
+	_PAD_CFG_STRUCT(GPP_B24, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_BUF(TX_RX_DISABLE), 0),
+
+	/* GPP_B25 - GSPI1_CLK_LOOPBK */
+	/* DW0: 0x40000700, DW1: 0x00000000 */
+	/* DW0: PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	/* PAD_CFG_NF(GPP_B25, NONE, DEEP, NF1), */
+	_PAD_CFG_STRUCT(GPP_B25, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_BUF(TX_RX_DISABLE), 0),
+
+	/* ------- GPIO Group GPP_T ------- */
+
+	/* GPP_T0 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	PAD_NC(GPP_T0, NONE),
+
+	/* GPP_T1 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	PAD_NC(GPP_T1, NONE),
+
+	/* GPP_T2 - FUSA_DIAGTEST_EN */
+	/* DW0: 0x44000b00, DW1: 0x00001000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	/* PAD_CFG_NF(GPP_T2, DN_20K, DEEP, NF2), */
+	_PAD_CFG_STRUCT(GPP_T2, PAD_FUNC(NF2) | PAD_RESET(DEEP) | PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE), PAD_PULL(DN_20K)),
+
+	/* GPP_T3 - FUSA_DIAGTEST_MODE */
+	/* DW0: 0x44000b00, DW1: 0x00001000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	/* PAD_CFG_NF(GPP_T3, DN_20K, DEEP, NF2), */
+	_PAD_CFG_STRUCT(GPP_T3, PAD_FUNC(NF2) | PAD_RESET(DEEP) | PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE), PAD_PULL(DN_20K)),
```

Comment:
All GPP_T should probably be NC, they are not present o the schematics. And probably the whole GPP_T group should be removed from this file.

Reply:
[Removed](https://github.com/Dasharo/coreboot/pull/576/commits/68fe32358db1c4826321169730cb09ec89ec4a40#diff-2f9e6fbf9293a4dbeecc2314eb0b449c4eb64691bfc1c58981fc506a602441bcL135)